tests/data_group_test.py

```diff
@@ -124,6 +124,12 @@ def test_setitem_str_key_replace():
     assert sc.identical(dg['a'], sc.scalar(2))
 
 
+def test_setitem_nonstr_key_fails():
+    dg = sc.DataGroup(a=sc.arange('x', 2))
+    with pytest.raises(ValueError):
+        dg[0] = sc.scalar(1)
```

Comment:
Can you add another test, verifying that this is not supported in `__init__` either (unless we have that elsewhere already)?

Reply:
Yes that test already exists [here](https://github.com/scipp/scipp/blob/740f9580c6f606a79f1f651eba55407a5c3313a5/tests/data_group_test.py#L31).